app/src/main/java/com/tomtom/sdk/examples/navigation/NavigationExamplesActivity.kt

```diff
@@ -11,15 +11,135 @@
+import com.tomtom.sdk.vehicle.Vehicle
 
 class NavigationExamplesActivity : AppCompatActivity() {
+
     private lateinit var binding: ActivityNavigationExamplesBinding
+    private lateinit var routePlanner: RoutePlanner
+    private lateinit var routePlanningOptions: RoutePlanningOptions
+    private lateinit var tomTomNavigation: TomTomNavigation
+    private lateinit var locationProvider: LocationProvider
+    private lateinit var route: Route
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         binding = ActivityNavigationExamplesBinding.inflate(layoutInflater)
         setContentView(binding.root)
+        planRoute()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        tomTomNavigation.close()
+        locationProvider.close()
+    }
+
+    private fun planRoute() {
+        routePlanner = OnlineRoutePlanner.create(context = this, apiKey = TOMTOM_API_KEY)
+        val amsterdam = GeoPoint(latitude = 52.37616, longitude = 4.90828)
+        val rotterdam = GeoPoint(latitude = 51.90546, longitude = 4.46662)
+        val itinerary = Itinerary(origin = amsterdam, destination = rotterdam)
+        routePlanningOptions = RoutePlanningOptions(
+            itinerary = itinerary,
+            guidanceOptions = GuidanceOptions(progressPoints = ProgressPoints.All),
+            vehicle = Vehicle.Car()
+        )
+        routePlanner.planRoute(
+            routePlanningOptions,
+            object : RoutePlanningCallback {
+                override fun onSuccess(result: RoutePlanningResponse) {
+                    route = result.routes.first()
+                    startNavigation()
+                }
+
+                override fun onFailure(failure: RoutingFailure) {
+                    Log.e(TAG, "Unable to calculate a route: " + failure.message)
+                }
+
+                override fun onRoutePlanned(route: Route) = Unit
+            }
+        )
+    }
+
+    private fun createLocationProvider() {
+        val routeGeoLocations = route.geometry.map { GeoLocation(it) }
+        val simulationStrategy = InterpolationStrategy(routeGeoLocations)
+        locationProvider = SimulationLocationProvider.create(strategy = simulationStrategy)
+        locationProvider.enable()
+    }
+
+    private fun createNavigationConfiguration(): Configuration {
+        val navigationTileStore = NavigationTileStore.create(
+            context = this,
+            navigationTileStoreConfig = NavigationTileStoreConfiguration(apiKey = TOMTOM_API_KEY)
+        )
+        val routeReplanner = OnlineRouteReplannerFactory.create(routePlanner)
+        createLocationProvider()
+        return Configuration(
+            context = this,
+            locationProvider = locationProvider,
+            navigationTileStore = navigationTileStore,
+            routeReplanner = routeReplanner
+        )
+    }
+
+    private fun startNavigation() {
```

Comment:
Could we put `startNavigation` right after `planRoute` (since that's where we ultimately call it from)? 
As well as:
- `createNavigationConfiguration` after `startNavigation`
- `createLocationProvider` after `createNavigationConfiguration`.

I think it would  be easier to follow that way.

Reply:
`startNavigation` depends on callback result from `planRoute` so it is not possible to put one after another, but others is possible, refactored a little bit.

Reply:
I meant the function _definitions_, not the function _calls_ (obviously). For better readability, you can place the function definitions after/below the point where you have the function call.